doc/adrs/095_simplify_the_manipulation_of_nodes_by_switching_to_a_common_box_model.adoc

```diff
@@ -0,0 +1,458 @@
+
+A list layout with the direction `row` will have to use the rule `align-items: stretch`.
+As a result, all the children will have the same height which will be the height of the content box of their parent.
+In a similar manner, in a list layout with the direction `column`, the rule `justify-items: stretch` will have to be used.
+In this case, all the children will have the same width which will be the width of the content box of their parent.
+
+Here are some examples use cases from various projects:
+
+The following configuration could be used to create a class from a class diagram for example:
+
+```
+direction: column
+align-items: start
+justify-items: stretch
+```
+
+NOTE: Insert a demo
+
+In this case, the footprint of all the children will have the same width, since they will be stretched on the row axis, using the width of the footprint of the largest child, but they will each keep their own height.
+They will simply be laid out from the top of the content box since the option `align-items` is using the value `start`.
+
+
+On the other hand, the following configuration could be use for an UML state machine which would require child nodes occupying all the available space of the parent at all time:
+
+```
+direction: column
+align-items: stretch
+justify-items: stretch
+```
+
+NOTE: Insert a demo
+
+In this situation, all the child nodes will have the same width, the width of the footprint of the larget child, but the content box of their parent will have for height, the sum of the height of the children.
+
+
+In order to support a vertical layout of the children the following configuration could instead be used:
+
+```
+direction: row
+align-items: stretch
+justify-items: stretch
+```
+
+NOTE: Insert a demo
+
+
+==== Free form layout strategy
+
+===== Overlap
+
+If a new node were to overlap existing nodes, we will move the existing nodes in one direction and one direction only to have enough space to insert the new node.
+This direction can only be verticaly or horizontaly.
+As such, moving a node to make room for another one will be done with a transalation changing either the parameter `x` or `y` of its position.
+
+If the newly moved nodes are also overlapping other nodes, we will move these nodes in one direction to have enough space.
+We will continue to do so recursively.
+Since a diagram has no bounds in any direction and since a node can be as big as necessary, moving existing nodes in this way can be done recursively without issues.
+
+NOTE: Insert a demo
+
+=== Inside label box
+
+The inside label box is a place where labels can be put inside of the node.
+These labels will count torward the size and thus the footprint of the node.
+Child nodes can only be layouted with a position inside the content box.
+After child nodes have been layouted, the full size of the content box will be known.
+Then we will compute the size of the label box, as a result a child node will never be able to be placed in the inside label box.
```

Comment:
Shouldn't it be the inverse? (label box computed before the child nodes)

Reply:
The label box is encapsulating the content box (which contains the child nodes). So we will compute the content box and after that, when we will have its size, we will compute the label box which will surround it